crates/sui-types/src/nitro_attestation.rs

```diff
@@ -594,8 +594,30 @@ impl AttestationDocument {
                             pcr_vec.push(value.to_vec());
                         }
                     }
+
+                    // valid key is 0..31, can parse with u8.
+                    let key_u8 = u8::try_from(key).map_err(|_| {
```

Comment:
we already defined it above

Reply:
yeah but i changed it from u64 to u8 since the valid values can be 0..31 only!

Reply:
so you probably want to guard that try_from with is_upgraded_parsing as well, no?

Reply:
yes😅all error paths here should be guarded.